src/app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,30 @@
+class LikesController < ApplicationController
+  before_action :set_user
+  before_action :set_function
+
+  def create
+    @like = @function.likes.new(user: current_user)
+    raise Unauthorized unless can?(current_user, @like, action_name.to_sym)
+
+    @like.save
+    redirect_back fallback_location: [@user, @function]
+  end
+
+  def destroy
+    @like = @function.likes.find_by(user: current_user) || raise(ActiveRecord::RecordNotFound)
+    raise Unauthorized unless can?(current_user, @like, action_name.to_sym)
```

Comment:
will it be better if you get action name inside `can?` method not passing it as a parameter ?

Reply:
that's a good suggestions, I think I'm always using this `can?` method with current_user, so yeah I will do this, but instead of doing it in this PR I'll refactor every use for this `can?` in a separate PR, just not to mix refactoring with this PR, it's already big